scripts/run_clippy.sh

```diff
@@ -1,14 +0,0 @@
-#!/usr/bin/env bash
```

Comment:
discussed offline: This logic was added as a rust unit test in #9608, this copy is now redundant. 

@nagisa I think it's still quite convenient to have clippy in a script. One idea to retain that would be to change that script to just call cargo nextest for that test. The drawback is that the script may get out of date but it would be smoother transition for anyone used to using that script. WDYT?

Reply:
Done.